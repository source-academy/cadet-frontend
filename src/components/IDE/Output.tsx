import * as React from 'react'

import { TextArea } from '@blueprintjs/core'

import { InterpreterOutput, ResultOutput } from '../../reducers/states'

export interface IOutputProps {
  output: InterpreterOutput[]
}

const Output: React.SFC<IOutputProps> = props => (
  <TextArea
    value={JSON.stringify((props.output.slice(-1)[0] as ResultOutput).value)}
    disabled={true}
    className="col-xs-6"
  />
)

export default Output
