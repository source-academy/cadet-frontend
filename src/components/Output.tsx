import * as React from 'react'

import { TextArea } from '@blueprintjs/core'

import { InterpreterOutput, ResultOutput } from '../reducers/playground'

export interface IOutputProps {
  output: InterpreterOutput[]
}

const Output: React.SFC<IOutputProps> = props => (
  <TextArea
    value={JSON.stringify((props.output.slice(-1)[0] as ResultOutput).value)}
    disabled={true}
  />
)

export default Output
