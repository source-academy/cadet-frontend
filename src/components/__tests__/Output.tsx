import * as React from 'react'

import { shallow } from 'enzyme'

import Output from '../Output'

test('Output renders correctly', () => {
  const props = {
    output: ['abc', 'def', 'ghi']
  }
  const app = <Output {...props} />
  const tree = shallow(app)
  expect(tree.debug()).toMatchSnapshot()
})
