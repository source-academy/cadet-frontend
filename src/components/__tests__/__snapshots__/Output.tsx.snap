// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output renders correctly 1`] = `"<Blueprint2.TextArea value=\\"\\"abc\\"\\" disabled={true} />"`;
